### PR TITLE
solution

### PR DESCRIPTION
**Pull request by maksym-kostetskyi**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -16,7 +16,7 @@
   "license": "GPL-3.0",
   "devDependencies": {
     "@mate-academy/eslint-config": "*",
-    "@mate-academy/scripts": "^0.3.5",
+    "@mate-academy/scripts": "^1.2.8",
```
Comment (SerhiiKirik) [inline package.json:19]: In the future - Do not add extra files to PR
To add the file to a commit use git add src/.

File: src/transformState.js [hunk 2]
```diff
@@ -5,7 +5,27 @@
  * @param {Object[]} actions
  */
 function transformState(state, actions) {
-  // write code here
+  for (let i = 0; i < actions.length; i++) {
```
Comment (SerhiiKirik) [inline src/transformState.js:8]: Let's use for of here
Also, use a switch case for this

**Review by SerhiiKirik — CHANGES_REQUESTED** [review]
Well done 🚀 
Let's improve your code 🧑‍💻

File: src/transformState.js [hunk 3]
```diff
@@ -5,7 +5,27 @@
  * @param {Object[]} actions
  */
 function transformState(state, actions) {
-  // write code here
+  for (const object of actions) {
```
Comment (DarkMistyRoom) [inline src/transformState.js:8]:
```suggestion
  for (const action of actions) {
```

File: src/transformState.js [hunk 4]
```diff
@@ -5,7 +5,27 @@
...
+    for (const value of Object.values(object)) {
```
Comment (DarkMistyRoom) [inline src/transformState.js:9]: why do you need this nested loop? you just need to check value of `action.type`

File: src/transformState.js [hunk 5]
```diff
@@ -5,7 +5,27 @@
...
+      switch (value) {
+        case 'addProperties':
+          for (const [key1, value1] of Object.entries(object.extraData)) {
+            state[key1] = value1;
+          };
```
Comment (DarkMistyRoom) [inline src/transformState.js:14]: I see that you really like loops, but would be more effectively to use `Object.assign` here

File: src/transformState.js [hunk 6]
```diff
@@ -5,7 +5,27 @@
...
+          break;
+        case 'removeProperties':
+          for (const key2 of object.keysToRemove) {
+            delete state[key2];
+          };
```
Comment (DarkMistyRoom) [inline src/transformState.js:19]: here it's ok)

File: src/transformState.js [hunk 7]
```diff
@@ -5,7 +5,27 @@
...
+          break;
+        case 'clear':
+          for (const key3 of Object.keys(state)) {
+            delete state[key3];
+          };
```
Comment (DarkMistyRoom) [inline src/transformState.js:24]: It is ok, but also you can just use `for in` to iterate through object keys

**Review by DarkMistyRoom — CHANGES_REQUESTED** [review]

**Review by mykhalenych — APPROVED** [review]

**Pull request closed without merging**